### PR TITLE
feat(accordion): add support to rendering validation icon beside heading FE-3031 

### PR DESCRIPTION
**Pull request by edleeks87**
### Proposed behaviour

Add support for rendering the `ValidationIcon` next to the `Accordion` heading by passing a string value to one of the `error`, `warning` or `info` props.

### Current behaviour

There is no support for rendering `ValidationIcon`.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context

image

image

image

image

image

image

### Testing instructions

Updates can be tested locally http://localhost:9001/?path=/story/design-system-accordion--validation-icon

### CODE REVIEW

File: src/components/accordion/accordion.component.js [hunk 1]
```diff
@@ -119,6 +124,7 @@ const Accordion = React.forwardRef(({
       >
         <StyledAccordionHeadingsContainer
           data-element='accordion-headings-container'
+          hasValidation={ showValidationIcon }
```
Comment (jamime) [inline src/components/accordion/accordion.component.js:127]:
```suggestion
          hasValidationIcon={ showValidationIcon }
```

File: src/components/accordion/accordion.stories.mdx [hunk 2]
```diff
@@ -177,6 +178,70 @@ import { Accordion, AccordionGroup } from "carbon-react/lib/components/accordion
...
+
+<Preview>
+  <Story name="validation icon">
+    {() => {
+      const [errors, setErrors] = useState({ one: 'error', two: 'error' });
+      const [warnings, setWarnings] = useState({});
+      const [infos, setInfos] = useState({});
+      const handleChange = (id, type, setter, msg) => {
+        const update = type[id] ? undefined : msg;
+        setter({ ...type, [id]: update });
```
Comment (jamime) [inline src/components/accordion/accordion.stories.mdx:193]:
```suggestion
        setter(previous => ({ ...previous, [id]: update }));
```

This is async and can cause issues.

File: src/components/accordion/accordion.stories.mdx [hunk 3]
```diff
@@ -177,6 +178,70 @@ import { Accordion, AccordionGroup } from "carbon-react/lib/components/accordion
...
+The `ValidationIcon` can be displayed in the `Accordion` header by passing in a string via one of the `error`, 
+`warning` or `info` props. This message will be displayed as in the tooltip.
+
+<Preview>
+  <Story name="validation icon">
+    {() => {
+      const [errors, setErrors] = useState({ one: 'error', two: 'error' });
+      const [warnings, setWarnings] = useState({});
+      const [infos, setInfos] = useState({});
+      const handleChange = (id, type, setter, msg) => {
```
Comment (jamime) [inline src/components/accordion/accordion.stories.mdx:191]:
```suggestion
      const handleChange = (id, setter, msg) => {
```

**Review by jamime — CHANGES_REQUESTED** [review]

File: src/components/accordion/accordion.stories.mdx (hunk 2 above)
Comment (edleeks87) [reply src/components/accordion/accordion.stories.mdx:193]: Yeah good spot thanks

File: src/components/accordion/accordion.stories.mdx (hunk 3 above)
Comment (edleeks87) [reply src/components/accordion/accordion.stories.mdx:191]: I need the type param here so I can do `const update = type[id] ? undefined : msg;`

**Comment by dawidzarzycki** [comment]
I didn't find nothing more than @jamime did, I'm happy to approve after `requested changes` will gone

**Review by jamime — APPROVED** [review]

**Review by dawidzarzycki — DISMISSED** [review]

**Review by dawidzarzycki — APPROVED** [review]

File: src/components/accordion/accordion.style.js [hunk 4]
```diff
@@ -48,8 +49,17 @@ const StyledAccordionIcon = styled(Icon)`
...
+  display: grid;
+  ${({ hasValidationIcon }) => (
+    hasValidationIcon ? 'grid-template-columns: auto auto;' : 'grid-template-rows: auto auto;'
+  )}
+  
+  ${ValidationIconStyle} {
+    height: 20px;
+    position: relative;
+    top: 2px;
+    z-index: 1000;
```
Comment (ianforster-sage) [inline src/components/accordion/accordion.style.js:61]: In base-theme.config.js we have a lookup for various zIndex values. Could you move this there, and also check how it renders if composed within a {{FullScreenDialog}}, for example.
https://github.com/Sage/carbon/blob/b998f5ca621a44be743341fc4ead8c71bc03e044/src/style/themes/base/base-theme.config.js#L314

Comment (edleeks87) [reply src/components/accordion/accordion.style.js:61]: Ah that's actually not needed, I meant to remove it so will just delete it

**Review by ianforster-sage — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

File: cypress/features/regression/designSystem/accordion.feature [hunk 5]
```diff
@@ -1,20 +1,29 @@
 Feature: Design System Accordion component
   I want to test Design System Accordion component
 
-  Background: Open Design System Accordion component page
-    Given I open "Design System Accordion" component page "default_story" in no iframe
+  # Background: Open Design System Accordion component page
```
Comment (DlgSHi) [inline cypress/features/regression/designSystem/accordion.feature:4]: Do we still need this ? :)

**Review by DlgSHi — DISMISSED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by ianforster-sage — APPROVED** [review]

**Review by chrisbarber86 — APPROVED** [review]

**Pull request merged**